presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -4974,13 +4977,13 @@ public void testInvalidType()
         computeActual("SELECT CAST(null AS array<foo>)");
     }
 
-    @Test(expectedExceptions = RuntimeException.class, expectedExceptionsMessageRegExp = "\\Qline 1:21: '+' cannot be applied to varchar, bigint\\E")
+    @Test(expectedExceptions = RuntimeException.class, expectedExceptionsMessageRegExp = "\\Qline 1:21: '+' cannot be applied to varchar(2147483647), bigint\\E")
```

Comment:
Currently variable argument functions do not play well with arguments using types parametrized.
The variable argument functions mechanism requires that all the arguments are of exactly same type. We cannot enforce that base must match but parameters are the same.

What could possible be done to change definition of `concat`